### PR TITLE
feat: add ESLint configuration for cypress-test-suite

### LINKED ISSUE #3536: Enforce ES-Lint rules for cypress-test-suite
Hi, I think as we are progressing with tests we need some kind of rules and standard notations to prevent us from writing bad code.
### Proposed solution

ES-Lint is pluggable and configurable linter tool for identifying and reporting on standard coding patterns in JavaScript. Cypress team has also created an ESLint extension for standard cypress rules https://github.com/cypress-io/eslint-plugin-cypress. I suggest incorporating ESLint for our cypress-test-suite
### Alternatives considered

Prettier but it only formats and beautifies code rather than actually complaining about coding patterns.

### PR DESCRIPTION
**Pull request by kushthedude**
- ESLint and Prettier won't have any conflicts now, I have fixed all conflicts as well as for safety there is an eslint-prettier-config package which would override any conflicts if present.
- In a later PR, I am planning to add a handy command like `yarn check` & `yarn lint` which will run both prettier and eslint linter on the code as well as fix the errors which can be fixed by automated linters respectively.
- A succeeding PR will be made soon to incorporate ESLint as a workflow job too in ui_job.
- I have used Google's as well as Cypress recommended configuration for our ESLint config.
fixes #3536

### CODE REVIEW

File: main/tests/cypress/cypress/integration/project/grid/misc/expressions.spec.js [hunk 1]
```diff
@@ -2,312 +2,286 @@
...
 
 /**
  * Utility method to fill something into the expression input
  * Need to wait for OpenRefine to preview the result, hence the cy.wait
  */
 function typeExpression(expression) {
-	cy.get('textarea.expression-preview-code').type(expression)
-	cy.wait(250)
+  cy.get('textarea.expression-preview-code').type(expression);
+  cy.get('body[ajax_in_progress="false"]');
```
Comment (kushthedude) [inline main/tests/cypress/cypress/integration/project/grid/misc/expressions.spec.js:14]: @fgiroud Cypress discourages us to specify custom wait in the spec; any workaround which you could suggest me here as this is failing the test? Or should we ignore the lint for just this line 🤔 ?

**Comment by fgiroud** [comment]
@kushthedude looks like some conflicts due to the previous PR

**Comment by kushthedude** [comment]
> @kushthedude looks like some conflicts due to the previous PR

should be fixed now

**Comment by kushthedude** [comment]
@fgiroud Its ready to go in
@wetneb can you please look into here as I didnt change any java code but coveralls is complaning ?

**Comment by wetneb** [comment]
@kushthedude the change of coverage is spurious, I will change the threshold on coveralls' side so that it is not reported as a failure.

**Review by fgiroud — APPROVED** [review]

**Pull request merged**